readthedocs/vcs_support/base.py

```diff
@@ -102,7 +103,13 @@ def run(self, *cmd, **kwargs):
             'shell': False,
         })
 
-        build_cmd = self.environment.run(*cmd, **kwargs)
+        try:
+            build_cmd = self.environment.run(*cmd, **kwargs)
+        except BuildEnvironmentWarning as e:
```

Comment:
It is supposed that we are not logging `BuildEnvironmentWarning` already (https://github.com/readthedocs/readthedocs.org/blob/70ca7c737a3db2539160e74f8ababaaf4743f30a/readthedocs/projects/tasks.py#L286). I'm not sure to follow why catching this exception and re-raising it as `RepositoyError` will avoid the log in Sentry.

I feel confused here :confused:

Reply:
Yes, but that isn't the cause. We are logging it manually here 

https://github.com/readthedocs/readthedocs.org/blob/00ab116ec09e75b74c20c1abdc7b9a9a47d9e673/readthedocs/projects/tasks.py#L254-L265

That try...except block only captures RepositoryError and LockTimeout, everything else is logged

Reply:
Gotcha! Do you know why the `sync_repo` is not ran inside the context manager that manages all the exceptions in the `__exit__` as we do run all the rest of the commands?

Reply:
It runs inside one https://github.com/readthedocs/readthedocs.org/blob/83e31f87c46845d11bfe6c7de8a417a6fdfb0c3f/readthedocs/projects/tasks.py#L449-L456

But again, that's not the problem. We still log the exception to sentry manually.

Reply:
I'm talking about this one which does not seem to be ran inside a context manager:

https://github.com/readthedocs/readthedocs.org/blob/83e31f87c46845d11bfe6c7de8a417a6fdfb0c3f/readthedocs/projects/tasks.py#L243

Reply:
Not really sure, we don't record that action as a build, so we don't need the all the stuff from the context manager. It doesn't seen related to this anyway. Both calls use the default env https://github.com/readthedocs/readthedocs.org/blob/83e31f87c46845d11bfe6c7de8a417a6fdfb0c3f/readthedocs/vcs_support/base.py#L67-L67 and don't use the context manager.

Reply:
Let's merge and see what happen after the next deploy :)